src/librustc_mir/const_eval/eval_queries.rs

```diff
@@ -94,10 +94,13 @@ pub(super) fn mk_eval_cx<'mir, 'tcx>(
     )
 }
 
-pub(super) fn op_to_const<'tcx>(
+#[derive(Debug)]
+pub(super) struct RefersToStatic;
```

Comment:
I don't think we need a new error type here. There are tons of errors that can happen in const-eval, why is this one so special that it has to shop up here? There is no precedent for anything like this.

Reply:
This is not actually an error. This is a signal that we should turn this into a `ByRef` constant instead of trying to use `Scalar` or `Slice`. At the very least it must be documented to explain this properly, but imo you an also just use `Option` as the return type of `try_op_to_const`